examples/kubernetes/1.10/cilium-ds.yaml

```diff
@@ -70,6 +71,11 @@ spec:
               configMapKeyRef:
                 name: cilium-config
                 key: disable-ipv4
+          - name: "SIDECAR_HTTP_PROXY"
+            valueFrom:
+              configMapKeyRef:
+                name: cilium-config
```

Comment:
Sorry, please verify one thing. If the users decide to use exactly this file (daemonset) to upgrade their cluster, and still using the old config map that does not contain the `sidecar-http-proxy` option, will cilium start? If it does not start, there's a field called `optional` and it should be used here `optional=true`

Reply:
Good call. The problem with that is that if we make this optional, the default value will be `""`, so the flag will become `--sidecar-http-proxy=`, which will enable sidecar by default. :/ I would prefer not to do this change if possible.

And I think that's OK?

Reply:
@rlenglet I spoke with @raybejjani  and I think we found a good solution for this:
```
diff --git a/daemon/main.go b/daemon/main.go
index 73f23ba03..20d3ae373 100644
--- a/daemon/main.go
+++ b/daemon/main.go
@@ -398,6 +398,7 @@ func init() {
                "restore", true, "Restores state, if possible, from previous daemon")
        flags.Bool("sidecar-http-proxy", false, "Disable host HTTP proxy, assuming proxies in sidecar containers")
        flags.MarkHidden("sidecar-http-proxy")
+       viper.BindEnv("sidecar-http-proxy", "CILIUM_SIDECAR_HTTP_PROXY")
        flags.BoolVar(&singleClusterRoute, "single-cluster-route", false,
                "Use a single cluster route instead of per node routes")
        flags.StringVar(&socketPath,
diff --git a/examples/kubernetes/templates/v1/cilium-ds.yaml.sed b/examples/kubernetes/templates/v1/cilium-ds.yaml.sed
index c6737f209..2e9c5081e 100644
--- a/examples/kubernetes/templates/v1/cilium-ds.yaml.sed
+++ b/examples/kubernetes/templates/v1/cilium-ds.yaml.sed
@@ -43,7 +43,6 @@ spec:
           - "--kvstore=etcd"
           - "--kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config"
           - "--disable-ipv4=$(DISABLE_IPV4)"
-          - "--sidecar-http-proxy=$(SIDECAR_HTTP_PROXY)"
         ports:
           - name: prometheus
             containerPort: 9090
@@ -71,11 +70,12 @@ spec:
               configMapKeyRef:
                 name: cilium-config
                 key: disable-ipv4
-          - name: "SIDECAR_HTTP_PROXY"
+          - name: "CILIUM_SIDECAR_HTTP_PROXY"
             valueFrom:
               configMapKeyRef:
                 name: cilium-config
                 key: sidecar-http-proxy
+                optional: true
           # Note: this variable is a no-op if not defined, and is used in the
           # prometheus examples.
           - name: "CILIUM_PROMETHEUS_SERVE_ADDR"
```

With this changes it will allow for a seamless upgrade for users running the daemonset with their existing configmap

Reply:
I like that. I’ll update the PR. Thanks!

Reply:
Done.